### DISCUSSION TITLE
Add a zoomable component

### DISCUSSION THREAD

**Discussion by peterfritz** [mantinedev/mantine#4106] [Feature requests]
Hi there,

Would it be possible to add a zoomable component to Mantine? I've worked on a lot of projects that would've benefited from the existence of such a component but although I tried, I was never able to get it quite right.

I believe that this component would make an excellent addition to the "Other packages" section. However, I am uncertain about its complexity or usefulness to others.

Thank you for your time and consideration.

P.S. Let me know if there's anything I can do to help with this feature request. I would love to help.

**Comment by KilianB** [comment]
You could begin by describing by what exactly you mean by a zoomable component. Move or only pinch interaction? Are we only talking about images or all kind of components? Is it a wrapper around child nodes?

**Reply by peterfritz** [reply]
Hi @KilianB,

Thank you for your message. By a zoomable component, I mean a component that can be zoomed in and out using interactions such as pinch, scroll, ctrl + scroll, or click, and, after zoomed, can be panned using scroll, shift + scroll, or drag. It would be amazing if a wrapper could be used to enable this behavior for almost every component, but I'm not sure if it's possible given differences in components' styles.

Regarding examples, I couldn't find many links to share, but here are two that demonstrate the behavior.

The first one implements pinch, scroll, and double click to zoom, and drag to pan:
https://www.easyzoom.com/embed/93940
(Note that the image optimization should be ignored.)

The second example is from Google Photos, which allows for scroll and double click to zoom, and drag to pan, but pinch interactions may not work correctly:
https://photos.google.com/share/AF1QipMpJu2q0ePUhAawt9uefXNATYOgIcEgSQI1jostLqMgwmf1H4T3ww2N4wmFz237cQ/photo/AF1QipPD8WvbyR3jp3ail057IAriuh82lNTNBjQMDC6q?key=T3hXcHBRb2hJbDREMkpublk0cXpxV2c0UVFhU1J3

I hope these examples help to give you a better idea of the behavior I'm referring to. Let me know if you have any further questions or concerns.
